Presentations/04-descriptive-analytics/04-descriptive-analytics.Rmd

```diff
@@ -0,0 +1,919 @@
+              `Total Population` = sum(pop))
+```
+
+```{r, echo = FALSE, message = F}
+  census %>% 
+  group_by(region) %>%
+  summarise(`Number of States` = n_distinct(state),
+            `Total Population` = sum(pop),
+            `Average Population` = mean(pop),
+            `SD of Population` = sd(pop)) %>%
+  kable
+```
+
+---
+# Aggregating observations
+
+.exercise[
+### Exercise `r fa("keyboard")`
+Recreate the `region_stats` data set,  now including the average and the standard deviation of the population.
+]
+
+.can-edit[
+```{r, eval = F}
+  region_stats <- 
+    census %>% 
+    group_by(region) %>%
+    summarise(`Number of States` = n_distinct(state),
+              `Total Population` = sum(pop))
+```
+]
+
+---
+
+# Aggregating observations
+
+```{r, message = F}
+  region_stats <- 
+    census %>% 
+    group_by(region) %>%
+    summarise(`Number of States` = n_distinct(state),
+              `Total Population` = sum(pop),
+              `Average Population` = mean(pop),
+              `SD of Population` = sd(pop))
+```
+
+```{r, echo = FALSE, message = F}
+  region_stats%>%
+  kable
+```
+
+---
+
+.exercise[
+### Exercise `r fa("keyboard")`
+Use `huxtable` to format and export the object `region_stats`.
+]
+
+--
+
+```{r, echo = F}
+region_stats_table <-
+    region_stats %>%
+    rename(Region = region) %>%
+    as_hux %>%
+    set_header_cols("Region", TRUE)  %>% 
+    theme_bright()
+
+number_format(region_stats_table) <- "%6.0f"
+region_stats_table
+```
+
+---
+
+```{r}
+  region_stats_table <-
+    region_stats %>%
+    rename(Region = region) %>%
+    as_hux %>%
+    set_header_cols("Region", TRUE)  %>% 
+    theme_bright()
+  
+  quick_xlsx(region_stats_table,
+           file = file.path(rawOutput, "region-stats.xlsx"))
+  
+  quick_latex(region_stats_table,
+             file = file.path(rawOutput, "region-stats.tex"))
+  
+```  
+
+---
+
+class: inverse, center, middle
+name: sec5
+
+# Ok, can we run some regressions now?!
+
+<html><div style='float:left'></div><hr color='#D38C28' size=1px width=1100px></html>
+
+---
+
+# Running regressions
+
+The base R command for linear regressions is called `lm`
+
+.command[
+## lm(formula, data, subset, weights, ...)
+
+- **formula:**	an object of class "formula" containing a symbolic description of the model
+- **data:** a data frame containing the variables indicated in the formula
+- *subset:* an optional vector specifying a subset of observations to be used in the regression
+- *weights:* an optional vector of weights to be used in the regression
+]
+
+Formulas can take three specifications:
+- `y ~ x1 + x2` regresses variable `y` on covariates `x1` and `x2`
+- `y ~ x1:x2` regresses variable `y` the interaction of covariates `x1` and `x2`
+- `y ~ x1*x2` is equivalent to `y ~ x1 + x2 + x1:x2`
+
+---
+
+# Running regressions
+
+.exercise[
+### Exercise `r fa("keyboard")`
+Using the `census` data, run a regression of population, urban population and number of marriages. 
```

Comment:
the outputs shown later for the solution of this exercise show a regression of divorces on population, urban population and number of marriages. Shouldn't the prompt for the exercise be: _"run a regression of divorces on population, urban population and number of marriages"_?

Reply:
this also applies for the next two slides and the next exercise (FE regression)